globus_action_provider_tools/flask/helpers.py

```diff
@@ -212,16 +212,19 @@ def pydantic_input_validation(
 except ImportError:
     # Flask < 2.2: Use the deprecated JSON encoder interface.
     json_provider_available = False
-    JsonProvider: Optional["DefaultJSONProvider"] = None
+
+    JsonProvider: type[DefaultJSONProvider] | None = None
```

Comment:
Wut. How does `type[DefaultJSONProvider]` not trigger a NameError? I obviously have more to learn...

I ran some quick test code and discovered that the following code WILL raise NameError _unless `from __future__ import annotations` is present.

```python
try:
    from bogus import nonsense
except ImportError:
    Demo: type[nonsense] | None = None
else:
    Demo = "the real thing goes here"
```

That's true going all the way back to Python 3.7 (3.7.13, at least, on my machine).

Reply:
A more basic example:

```python
instance: NoProblemsHere  # NameError!

from __future__ import annotations

instance: NoProblemsHere  # And indeed there are no problems!
```

Reply:
Yeah, this is `__future__` magic.

I'm not sure if you've followed some of the threads about annotations maybe becoming a data descriptor, but the implications are odd. With deferred annotations pulled from 3.11, we might be in an unprecedented situation in which there is a future import _which will never become the future behavior_. I may need to rethink how aggressively I lean on the future import behavior in my code.

Reply:
> there is a future import which will never become the future behavior

That's the exact plot of "Terminator: Dark Fate"! Is James Cameron a Python release manager?

Reply:
Jokes aside, this is all very surprising. I don't have an opinion about what _should_ happen in the future, only that this broke my expectations.